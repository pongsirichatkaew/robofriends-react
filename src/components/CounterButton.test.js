import { shallow, mount, render } from 'enzyme';
import CounterButton from './CounterButton';
import React from 'react';
import toJson from 'enzyme-to-json';

it('expect to render Counter Button', () => {
  const mockColor = 'red';
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it('correctly increment the counter', () => {
  const mockColor = 'red';
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.props().color).toEqual('red')
});
