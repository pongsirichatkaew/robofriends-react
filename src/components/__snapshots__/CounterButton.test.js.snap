// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Counter Button 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
